airflow/providers/amazon/aws/transfers/s3_to_redshift.py

```diff
@@ -112,6 +119,17 @@ def execute(self, context):
             copy_options=copy_options,
         )
 
+        if self.truncate_table:
+            truncate_statement = f'TRUNCATE TABLE {self.schema}.{self.table};'
```

Comment:
Does this allow users to create SQL injection?

Reply:
I think it is not different to the copy command for example (i.e. any direct command which is configurable via parameters).

How about: https://www.psycopg.org/docs/sql.html ? This way it will be first validated and then executed.

@turbaszek How do you prevent that in case of BigQuery for example?

Reply:
> How about: https://www.psycopg.org/docs/sql.html ? This way it will be first validated and then executed.

I think this sounds good, any escape checking will help.

> @turbaszek How do you prevent that in case of BigQuery for example?

In case of BQ everywhere where query is executed, it has to be provided by users. As far as I know (and I screened the code now) we do not construct queries anywhere.

Reply:
I am open to use psycopg to address the issue. 

Nevertheless, the only way I think someone could make SQL Injection is setting schema or table like:
`table/schema; DROP ALL TABLES OR WHATEVER`
If you don't specify any table, the command will fail, so wouldn't it be easier just to make strip of semicolon for table and schema? WTDY?

Reply:
Looking at 107-111 it seems common practise to do queries like this... there are 7 parameters used to construct the query 🙄  I think we should not risk any SQL injection but I'm ok with fixing this in a follow up PR or creating an issue for that so we do not forget about it